### PR TITLE
Tooltip Atom / Action

### PR DESCRIPTION
**Pull request by 0c370t**

### CODE REVIEW

File: clients/web/src/lib/components/actions/Tooltip/Tooltip.story.svelte [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            <span class="ml-8">With Arrow</span>
+            <input bind:checked={params.withArrow} type="checkbox" />
+        </div>
+    </svelte:fragment>
+    <Hst.Variant title="Default">
+        <div class="w-screen h-screen flex justify-center items-center">
+            {#key params}
+                <button class="border border-primary rounded-lg px-2 py-1 text-gray-200" use:tooltip={params}
+                    >{params.position}</button
+                >
```
Comment (zachstence) [inline clients/web/src/lib/components/actions/Tooltip/Tooltip.story.svelte:31]: Funky formatting here

File: clients/web/src/lib/components/actions/Tooltip/Tooltip.story.svelte [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            </select>
+        </div>
+        <div>
+            <span class="ml-8">With Arrow</span>
+            <input bind:checked={params.withArrow} type="checkbox" />
+        </div>
+    </svelte:fragment>
+    <Hst.Variant title="Default">
+        <div class="w-screen h-screen flex justify-center items-center">
+            {#key params}
```
Comment (zachstence) [inline clients/web/src/lib/components/actions/Tooltip/Tooltip.story.svelte:28]: What does this `{#key}` do? Never seen this before

File: clients/web/src/lib/components/actions/Tooltip/useTooltip.action.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import type {TooltipParams} from "./types";
+import "./tooltip.postcss";
+
+export const tooltip = (node: HTMLElement, {content, position = "top", withArrow}: TooltipParams) => {
+    const tooltip = document.createElement("div");
+    tooltip.classList.add("tooltip");
+    console.log({position, withArrow});
```
Comment (zachstence) [inline clients/web/src/lib/components/actions/Tooltip/useTooltip.action.ts:7]: Remove log

File: clients/web/src/lib/components/actions/Tooltip/useTooltip.action.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    node.addEventListener("mouseenter", renderTooltip);
+    node.addEventListener("mouseleave", removeTooltip);
+
+    return {
+        destroy() {
+            node.removeEventListener("mouseenter", renderTooltip);
+            node.removeEventListener("mouseleave", removeTooltip);
+        },
+    };
```
Comment (zachstence) [inline clients/web/src/lib/components/actions/Tooltip/useTooltip.action.ts:62]: What is this syntax? Is this returning an object with a function `destroy`? No function keyword needed?

File: clients/web/src/lib/components/actions/Tooltip/tooltip.postcss [hunk 5]
```diff
@@ -0,0 +1,25 @@
+.tooltip {
```
Comment (zachstence) [inline clients/web/src/lib/components/actions/Tooltip/tooltip.postcss:1]: Are these styles global? If so, we may need a more unique class name than `tooltip`. I could see that being duplicated somewhere else in the app

**Review by zachstence — CHANGES_REQUESTED** [review]
`src/lib/components/atoms/Tooltip/index.ts` is an empty file

File: clients/web/src/lib/components/actions/Tooltip/Tooltip.story.svelte (hunk 1 above)
Comment (0c370t) [reply clients/web/src/lib/components/actions/Tooltip/Tooltip.story.svelte:31]: This is what prettier does in some cases 😞 
I'll try to tweak it

File: clients/web/src/lib/components/actions/Tooltip/Tooltip.story.svelte (hunk 2 above)
Comment (0c370t) [reply clients/web/src/lib/components/actions/Tooltip/Tooltip.story.svelte:28]: Forces a re-render when the prop changes; in this case it forces a rerender when parameters change, I got funky behavior without it

File: clients/web/src/lib/components/actions/Tooltip/useTooltip.action.ts (hunk 3 above)
Comment (0c370t) [reply clients/web/src/lib/components/actions/Tooltip/useTooltip.action.ts:7]:
```suggestion
```

File: clients/web/src/lib/components/actions/Tooltip/useTooltip.action.ts (hunk 4 above)
Comment (0c370t) [reply clients/web/src/lib/components/actions/Tooltip/useTooltip.action.ts:62]: Function keyword isn't technically ever needed.

This does return that object; see https://svelte.dev/docs#template-syntax-element-directives-use-action

File: clients/web/src/lib/components/actions/Tooltip/tooltip.postcss (hunk 5 above)
Comment (0c370t) [reply clients/web/src/lib/components/actions/Tooltip/tooltip.postcss:1]: Not sure what other context `tooltip` would be showing up in, but I believe this is global

**Review by zachstence — APPROVED** [review]

**Pull request merged**